packages/cloud_firestore/cloud_firestore_web/lib/src/document_reference_web.dart

```diff
@@ -27,44 +26,29 @@ class DocumentReferenceWeb extends DocumentReferencePlatform {
 
   @override
   Future<void> set(Map<String, dynamic> data, [SetOptions? options]) async {
-    try {
-      await _delegate.set(
-        CodecUtility.encodeMapData(data)!,
-        convertSetOptions(options),
-      );
-    } catch (e) {
-      throw getFirebaseException(e);
-    }
+    return guard(() => _delegate.set(
```

Comment:
nit: In general, I would not use `=>` unless the content can be expressed in a single line.
So here I'd use `{}`